Move implementation for methods using inline-declared methods out from headers

csaavedra opened the pull request:
#### 59cd2096d26ab40a0e1319f3d38c469a0cdc7d57
<pre>
Move implementation for methods using inline-declared methods out from headers
<a href="https://bugs.webkit.org/show_bug.cgi?id=296934">https://bugs.webkit.org/show_bug.cgi?id=296934</a>

Reviewed by Ross Kirsling.

There are many, often trivial, methods implemented
in header files that are using methods from other classes
that are defined in a Inlines.h header file. The convention
in WebKit (enforced by the style-checker) is that one
should never include Inlines.h files in a header, as it has
a negative impact in build time.

The problem with this status-quo is that this obscures
the fact that those Inlines.h files are not always
included when needed, particularly in non-unified builds, but
also when the unified compilation units change. This causes
compilation warnings and/or of the likes of:

    error: inline function &apos;WebCore::ContextDestructionObserver::scriptExecutionContext&apos; is not defined

Since we don&apos;t want to include Inlines.h in headers,
the correct usage of the methods defined in those files
is to not use them in headers, but to have the implementations
using them defined in the CPP files instead.

This patch is an attempt to remedy this situation but
it&apos;s not exhaustive, but it fixes a lot of those errors.
Notably, it also modifies CodeGeneratorJS.pm to fix
this issue in generated-code as well.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::scriptExecutionContext const):
* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::scriptExecutionContext const):
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
(WebCore::MediaDevices::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/MediaStream.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.cpp:
(WebCore::RTCDtlsTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::RTCRtpSFrameTransform::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/webcodecs/WebCodecsBase.cpp:
(WebCore::WebCodecsBase::scriptExecutionContext const):
* Source/WebCore/Modules/webcodecs/WebCodecsBase.h:
* Source/WebCore/Modules/webdatabase/DatabaseContext.cpp:
(WebCore::DatabaseContext::document const):
* Source/WebCore/Modules/webdatabase/DatabaseContext.h:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(GenerateCallbackHeaderContent):
(GenerateCallbackImplementationContent):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunction.cpp:
(WebCore::JSTestCallbackFunction::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunction.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionGenerateIsReachable.cpp:
(WebCore::JSTestCallbackFunctionGenerateIsReachable::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionGenerateIsReachable.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithThisObject.cpp:
(WebCore::JSTestCallbackFunctionWithThisObject::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithThisObject.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithTypedefs.cpp:
(WebCore::JSTestCallbackFunctionWithTypedefs::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithTypedefs.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithVariadic.cpp:
(WebCore::JSTestCallbackFunctionWithVariadic::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithVariadic.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackInterface.cpp:
(WebCore::JSTestCallbackInterface::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackInterface.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackWithFunctionOrDict.cpp:
(WebCore::JSTestCallbackWithFunctionOrDict::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackWithFunctionOrDict.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestVoidCallbackFunction.cpp:
(WebCore::JSTestVoidCallbackFunction::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestVoidCallbackFunction.h:
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::ViewTransition::document const):
* Source/WebCore/dom/ViewTransition.h:
(WebCore::ViewTransition::document const): Deleted.
* Source/WebCore/html/MediaController.cpp:
(WebCore::MediaController::scriptExecutionContext const):
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/track/TrackListBase.cpp:
(WebCore::TrackListBase::scriptExecutionContext const):
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/page/IntersectionObserver.cpp:
(WebCore::IntersectionObserver::trackingDocument const):
* Source/WebCore/page/IntersectionObserver.h:
(WebCore::IntersectionObserver::trackingDocument const): Deleted.
* Source/WebCore/rendering/LegacyInlineBox.cpp:
(WebCore::LegacyInlineBox::lineStyle const):
* Source/WebCore/rendering/LegacyInlineBox.h:
(WebCore::LegacyInlineBox::lineStyle const): Deleted.
* Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp:
(WebCore::RenderMathMLFraction::element const):
* Source/WebCore/rendering/mathml/RenderMathMLFraction.h:
* Source/WebCore/rendering/mathml/RenderMathMLSpace.cpp:
(WebCore::RenderMathMLSpace::element const):
* Source/WebCore/rendering/mathml/RenderMathMLSpace.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
(WebCore::BackgroundFetchRegistration::scriptExecutionContext const):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.h:
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
(WebCore::XMLHttpRequestProgressEventThrottle::protectedTarget):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.h:
(WebCore::XMLHttpRequestProgressEventThrottle::protectedTarget): Deleted.

Canonical link: <a href="https://commits.webkit.org/298326@main">https://commits.webkit.org/298326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/47e19c3a2994f125510222b0087572ed45b55c35

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114800 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34545 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25009 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120962 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65488 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116689 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35174 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43103 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87254 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42123 "Build was cancelled. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Running run-layout-tests-in-stress-mode; layout-tests (exception)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117748 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28023 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103084 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67644 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27208 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21205 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64630 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97407 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21319 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124163 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41804 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31231 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96063 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42182 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99273 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95849 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24445 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41030 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18868 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37858 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41685 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47199 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41247 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44562 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42991 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling reviewed (approved):
Looks like a nice improvement; Inlines.h-related errors upon unified build shifts are truly the worst.
I wish this were easier to enforce with a style check (...but then we wouldn't even need that if we had a non-unified bot instead 🥲).

csaavedra committed:
Move implementation for methods using inline-declared methods out from headers
https://bugs.webkit.org/show_bug.cgi?id=296934

Reviewed by Ross Kirsling.

There are many, often trivial, methods implemented
in header files that are using methods from other classes
that are defined in a Inlines.h header file. The convention
in WebKit (enforced by the style-checker) is that one
should never include Inlines.h files in a header, as it has
a negative impact in build time.

The problem with this status-quo is that this obscures
the fact that those Inlines.h files are not always
included when needed, particularly in non-unified builds, but
also when the unified compilation units change. This causes
compilation warnings and/or of the likes of:

    error: inline function 'WebCore::ContextDestructionObserver::scriptExecutionContext' is not defined

Since we don't want to include Inlines.h in headers,
the correct usage of the methods defined in those files
is to not use them in headers, but to have the implementations
using them defined in the CPP files instead.

This patch is an attempt to remedy this situation but
it's not exhaustive, but it fixes a lot of those errors.
Notably, it also modifies CodeGeneratorJS.pm to fix
this issue in generated-code as well.

* Source/WebCore/Modules/WebGPU/GPUDevice.cpp:
(WebCore::GPUDevice::scriptExecutionContext const):
* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::scriptExecutionContext const):
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
(WebCore::MediaDevices::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/MediaStream.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::RTCDataChannel::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.cpp:
(WebCore::RTCDtlsTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::RTCRtpSFrameTransform::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/webcodecs/WebCodecsBase.cpp:
(WebCore::WebCodecsBase::scriptExecutionContext const):
* Source/WebCore/Modules/webcodecs/WebCodecsBase.h:
* Source/WebCore/Modules/webdatabase/DatabaseContext.cpp:
(WebCore::DatabaseContext::document const):
* Source/WebCore/Modules/webdatabase/DatabaseContext.h:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(GenerateCallbackHeaderContent):
(GenerateCallbackImplementationContent):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunction.cpp:
(WebCore::JSTestCallbackFunction::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunction.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionGenerateIsReachable.cpp:
(WebCore::JSTestCallbackFunctionGenerateIsReachable::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionGenerateIsReachable.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithThisObject.cpp:
(WebCore::JSTestCallbackFunctionWithThisObject::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithThisObject.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithTypedefs.cpp:
(WebCore::JSTestCallbackFunctionWithTypedefs::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithTypedefs.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithVariadic.cpp:
(WebCore::JSTestCallbackFunctionWithVariadic::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackFunctionWithVariadic.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackInterface.cpp:
(WebCore::JSTestCallbackInterface::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackInterface.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackWithFunctionOrDict.cpp:
(WebCore::JSTestCallbackWithFunctionOrDict::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestCallbackWithFunctionOrDict.h:
* Source/WebCore/bindings/scripts/test/JS/JSTestVoidCallbackFunction.cpp:
(WebCore::JSTestVoidCallbackFunction::scriptExecutionContext const):
* Source/WebCore/bindings/scripts/test/JS/JSTestVoidCallbackFunction.h:
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::ViewTransition::document const):
* Source/WebCore/dom/ViewTransition.h:
(WebCore::ViewTransition::document const): Deleted.
* Source/WebCore/html/MediaController.cpp:
(WebCore::MediaController::scriptExecutionContext const):
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/track/TrackListBase.cpp:
(WebCore::TrackListBase::scriptExecutionContext const):
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/page/IntersectionObserver.cpp:
(WebCore::IntersectionObserver::trackingDocument const):
* Source/WebCore/page/IntersectionObserver.h:
(WebCore::IntersectionObserver::trackingDocument const): Deleted.
* Source/WebCore/rendering/LegacyInlineBox.cpp:
(WebCore::LegacyInlineBox::lineStyle const):
* Source/WebCore/rendering/LegacyInlineBox.h:
(WebCore::LegacyInlineBox::lineStyle const): Deleted.
* Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp:
(WebCore::RenderMathMLFraction::element const):
* Source/WebCore/rendering/mathml/RenderMathMLFraction.h:
* Source/WebCore/rendering/mathml/RenderMathMLSpace.cpp:
(WebCore::RenderMathMLSpace::element const):
* Source/WebCore/rendering/mathml/RenderMathMLSpace.h:
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.cpp:
(WebCore::BackgroundFetchRegistration::scriptExecutionContext const):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRegistration.h:
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.cpp:
(WebCore::XMLHttpRequestProgressEventThrottle::protectedTarget):
* Source/WebCore/xml/XMLHttpRequestProgressEventThrottle.h:
(WebCore::XMLHttpRequestProgressEventThrottle::protectedTarget): Deleted.

Canonical link: https://commits.webkit.org/298326@main
